### PR TITLE
One dnn

### PR DESCRIPTION
**Pull request by indie**
# Description

Includes updates to README for better markdown-friendly TM&B, as well as some formatting and legal info updates.

Fixes # (github issue)

# Checklist

## All Submissions

- [ ] Do all unit and benchdnn tests (`make test` and `make test_benchdnn_*`)
      pass locally?
- [ ] Have you formatted the code using clang-format?

## New features

- [ ] Have you added relevant tests?
- [ ] Have you provided motivation for adding a new feature?

## Bug fixes

- [ ] Have you added relevant regression tests?
- [ ] Have you included information on how to reproduce the issue (either in a
      github issue or in this PR)?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -298,6 +297,11 @@ terms, and open source software license terms. These separate license terms
 govern your use of the third party programs as set forth in the
 "[THIRD-PARTY-PROGRAMS](THIRD-PARTY-PROGRAMS)" file.
 
---------
+# Legal Information
 
-[Legal Information](doc/legal_information.md)
```
Comment (vpirogov) [inline README.md:303]: Let me see what content is actually needed here. Meanwhile please revert the change.

File: README.md [hunk 2]
```diff
@@ -96,16 +96,16 @@ oneDNN supports platforms based on the following architectures:
...
-* 4th, 5th, 6th, 7th, and 8th generation Intel(R) Core(TM) processor
-* Intel(R) Xeon(R) processor E3, E5, and E7 family (formerly Sandy Bridge,
+* 4th, 5th, 6th, 7th, and 8th generation Intel&reg; Core&trade; processor
+* Intel&reg; Xeon&reg; processor E3, E5, and E7 family (formerly Sandy Bridge,
   Ivy Bridge, Haswell, and Broadwell)
-* Intel(R) Xeon Phi(TM) processor (formerly Knights Landing and Knights Mill)
-* Intel Xeon Scalable processor (formerly Skylake and Cascade Lake)
-* future Intel Xeon Scalable processor (code name Cooper Lake)
+* Intel&reg; Xeon Phi&trade; processor (formerly Knights Landing and Knights Mill)
+* Intel&reg; Xeon&reg; Scalable processor (formerly Skylake and Cascade Lake)
```
Comment (vpirogov) [inline README.md:103]: In this and some other places trademark symbols are not necessary, as the trademark was used with proper symol earlier in the text.

File: README.md [hunk 3]
```diff
@@ -96,16 +96,16 @@ oneDNN supports platforms based on the following architectures:
...
 The library is optimized for the following CPUs:
 * Intel Atom processor with Intel SSE4.1 support
-* 4th, 5th, 6th, 7th, and 8th generation Intel(R) Core(TM) processor
-* Intel(R) Xeon(R) processor E3, E5, and E7 family (formerly Sandy Bridge,
+* 4th, 5th, 6th, 7th, and 8th generation Intel&reg; Core&trade; processor
+* Intel&reg; Xeon&reg; processor E3, E5, and E7 family (formerly Sandy Bridge,
   Ivy Bridge, Haswell, and Broadwell)
-* Intel(R) Xeon Phi(TM) processor (formerly Knights Landing and Knights Mill)
-* Intel Xeon Scalable processor (formerly Skylake and Cascade Lake)
-* future Intel Xeon Scalable processor (code name Cooper Lake)
```
Comment (vpirogov) [inline README.md:104]: This line should stay.

File: README.md [hunk 4]
```diff
@@ -147,8 +147,8 @@ for standalone Intel MKL-DNN build.
 |:---                   |:---
 |lib/libmkldnn.so       | Intel MKL-DNN dynamic library
 |lib/libiomp5.so        | Intel OpenMP* runtime library
-|lib/libmklml_gnu.so      | Intel MKL small library for GNU* OpenMP runtime
-|lib/libmklml_intel.so    | Intel MKL small library for Intel(R) OpenMP runtime
+|lib/libmklml_gnu.so    | Intel MKL small library for GNU* OpenMP runtime
```
Comment (vpirogov) [inline README.md]: This looks like a commit from older version that is not relevant anymore.

File: README.md [hunk 5]
```diff
@@ -151,7 +150,7 @@ Intel C++ Compiler.
 
 ### GPU Engine
 
-Intel Processor Graphics is supported by the oneDNN GPU engine. The GPU
+Intel&reg; Processor Graphics are supported by the oneDNN GPU engine. The GPU
```
[the remark this answers is not in the document]
Comment (indie) [reply README.md]: Reworded to be more concise; thanks for the clarification.

File: README.md (hunk 2 above)
Comment (indie) [reply README.md]: :+1:

File: README.md (hunk 1 above)
Comment (indie) [reply README.md:303]: Done

**Comment by vpirogov** [comment]
Thanks, @indie!
